org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/rule/RuleHeaderPanel.java

```diff
@@ -28,49 +29,101 @@
 import org.eclipse.swt.widgets.Label;
 import org.sonarlint.eclipse.core.internal.utils.StringUtils;
 import org.sonarlint.eclipse.ui.internal.SonarLintImages;
-import org.sonarsource.sonarlint.core.commons.IssueSeverity;
-import org.sonarsource.sonarlint.core.commons.RuleType;
+import org.sonarsource.sonarlint.core.clientapi.backend.rules.AbstractRuleDto;
+import org.sonarsource.sonarlint.core.commons.SoftwareQuality;
 
-/** Rule header containing all information excluding the title and description */
+/**
+ *  Rule header containing all information excluding the title and description:
+ *  
+ *  Old Clean Code Taxonomy:
+ *  - rule type                           [label 1 & 2]
+ *  - rule severity                       [label 3 & 4]
+ *  
+ *  New Clean Code Taxonomy:
+ *  - clean code attribute                [label 1]
+ *  - software quality impact             [label 2 & 3]
+ *  - software quality impact (optional)  [label 4 & 5]
+ *  - software quality impact (optional)  [label 6 & 7]
+ *  
+ *  The rule key is shown in the old and new Clean Code Taxonomy!
+ */
 public class RuleHeaderPanel extends Composite {
-  private final Label ruleTypeIcon;
-  private final Label ruleTypeLabel;
-  private final Label ruleSeverityIcon;
-  private final Label ruleSeverityLabel;
+  private final Label label1;
+  private final Label label2;
+  private final Label label3;
+  private final Label label4;
+  private final Label label5;
+  private final Label label6;
+  private final Label label7;
```

Comment:
I think fields named like this are super hard to follow, and it's very easy to mix them up. If the panels have different structures, why not introduce 2 different classes? One `LegacyRuleHeaderPanel` (that will be deleted at some point) and this class that would be used for the new CCT?

Reply:
I see the point and I have no idea how to name them better. The issue with having two classes is that we would have to dispose of and create a new one from scratch every time we call `RuleDetailsPanel#updateRule(...)` based on the new CCT or the old CCT.
The **Rule Description view** is sadly already quite slow and flickery due to the disposing of the `RuleDescriptionPanel` every time we update it and I favor stable usability over maintainability in this case. Yes such a naming is hard to follow but with the description and it being a minor and simple class I think it is manageable.
WDYT?

Reply:
Another idea instead of adding/removing the view each time would be to just hide the one not in use: you could create the 2 classes, always add them to the tree and `setVisible` only one of them based on the CCT to use. 
The other argument is that the day we remove the legacy CCT, and I know it's not for tomorrow, it will be simpler to remove the legacy class.
I agree it's a tradeoff, I would still prefer maintainability and readability in this case over efficiency

Reply:
Resolved but was only possible by disposing of the `AbstractRuleHeaderPanel`. But this did not cause the big deal I was anticipating so no further decrease of performance on loading a rule description.